enhancements/alert-on-operator-updates.md

```diff
@@ -0,0 +1,158 @@
+services) running across a Kubernetes cluster. Given that the installed Operators have the freedom to do almost anything with appropriate access privileges, there is a 
+potential threat that these actions may cause harm to the cluster, are malicious, or detrimental to a cluster upgrade. 
+
+This enhancement deals with handling two such Operator actions:  
+- Alert is triggered when the operator deploys operands with an invalid [Pod Disruption Budget (PDB)](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget) that would cause an openshift upgrade issue
+- Alert is triggered when an operator tries to modify an existing on-board [Security Context Constraints (SCCs)](https://www.openshift.com/blog/understanding-service-accounts-sccs))
+ 
+
+### Goals
+
+1. Provide a mechanism to alert when an invalid PDB is being applied on to a cluster 
+2. Prevent operators from modifying existing SCCs in the cluster
+3. Provide a configuration option to cluster admins for a permissive mode vs. a enforcing mode
+    - Alert from the start and add enforcement after we get data on how many of these alerts are firing.
+    - Able to update independently what is an alerting rule and what is an enforcing rule. 
+    - Provide option to turn off an enforcing mode
+
+## Open Questions 
+
+1. Configuration option on OLM to make for a permissive mode vs. a enforcing mode 
+    - If there is an invalid PDB yaml being applied on to the cluster, shouldn't it be prevented altogether? Why have a permissive mode?
+   
+2. Validation library is used to define what components of an Operator are deemed potentially harmful 
+	 - How do I go about finding this? Also, for these components that are deemed harmful, what is the expected action to be taken? I
+	 presume they would need a different design strategy than the one for SCCs and PDB? 
+
+## Design Details
+
+The design is split into 3 sections, each one per goal defined above. 
+
+## 1. A mechanism to alert when an invalid PDB is being applied on to a cluster (a.k.a validating PDB yaml) 
+
+Let's consider an example PDB.yaml
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+
+PodDisruptionBudget has three fields:
+
+- A label selector `.spec.selector` to specify the set of pods to which it applies. This field is required.
+- `.spec.minAvailable` which specifies the number of pods that must still be available after the eviction, even in the absence of the evicted pod.
+- `.spec.maxUnavailable` which specifies the number of pods that can be unavailable after the eviction
+
+Note: `minAvailable` & `maxUnavailable` can be either an absolute number or a percentage.
+
+An PDB is considered invalid if any of these fields has incorrect values and gets applied on to a cluster.
```

Comment:
What exactly constitutes "incorrect"? Are you using that to mean the same as "invalid"? Defining those terms is central to this proposal.

Reply:
+1 as an operator author I want to know what OLM would consider incorrect (that would later need to be documented)